InvenTree/part/test_api.py

```diff
@@ -293,6 +294,113 @@ def test_invisible_chars(self):
+
+            url = reverse('api-part-category-detail', kwargs={'pk': cat_to_delete.id})
+
+            parts = []
+            # Create parts in the category to be deleted
+            for jj in range(3):
+                parts.append(Part.objects.create(
+                    name=f"Part xyz {jj}",
+                    description="Child part of the deleted category",
+                    category=cat_to_delete
+                ))
+
+            child_categories = []
+            child_categories_parts = []
+            # Create child categories under the category to be deleted
+            for ii in range(3):
+                child = PartCategory.objects.create(
+                    name=f"Child parent_cat {ii}",
+                    description="A child category of the deleted category",
+                    parent=cat_to_delete
+                )
+                child_categories.append(child)
+
+                # Create parts in the child categories
+                for jj in range(3):
+                    child_categories_parts.append(Part.objects.create(
+                        name=f"Part xyz {jj}",
+                        description="Child part in the child category of the deleted category",
+                        category=child
+                    ))
+
+            # Delete the created category (sub categories and their parts will be moved under the parent)
+            params = {}
+            if delete_parts:
+                params['delete_parts'] = '1'
+            if delete_child_categories:
+                params['delete_child_categories'] = '1'
+            response = self.delete(
+                url,
+                params,
+                expected_code=204,
+            )
+
+            self.assertEqual(response.status_code, 204)
+
+            if delete_parts:
+                if i == Target.delete_subcategories_delete_parts:
+                    # Check if all parts deleted
+                    self.assertEqual(Part.objects.count(), part_count_before)
+                elif i == Target.move_subcategories_to_parent_delete_parts:
+                    # Check if all parts deleted
+                    self.assertEqual(Part.objects.count(), part_count_before + len(child_categories_parts))
+            else:
+                # parts moved to the parent category
+                for part in parts:
+                    part.refresh_from_db()
+                    self.assertEqual(part.category, parent_category)
+
+                if delete_child_categories:
+                    for part in child_categories_parts:
+                        part.refresh_from_db()
+                        self.assertEqual(part.category, parent_category)
+
+            if delete_child_categories:
+                self.assertEqual(PartCategory.objects.count(), category_count_before)
+            else:
+                #  Check if all subcategories to parent moved to parent and all parts deleted
+                for child in child_categories:
```

Comment:
Suggestion: A short comment on why the assertations are made would be good for future debugging